app/src/main/java/com/example/task_planner_app/ui/dialog/DatePickerFragment.kt

```diff
@@ -0,0 +1,29 @@
+package com.example.task_planner_app.ui.dialog
+
+import android.app.DatePickerDialog
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.widget.DatePicker
+import androidx.fragment.app.DialogFragment
+import java.util.*
+
+class DatePickerFragment(
+    val listener: (day: Int, month: Int, year:Int) -> Unit
```

Comment:
Use a more concrete name for the listener. Perhaps dateSelectedListener so you know what this listener does from reading the code.